plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -77,6 +136,102 @@ protected String getStorageMetadataLocation(String materializedViewName)
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        ZonedDateTime lastFreshTime = (ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue();
+        assertThat(lastFreshTime).isNotNull();
+        int result3 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result3).isNotEqualTo(result2);  // mv is not stale anymore so all selects until next refresh returns same result
+        int result4 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        int result5 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result4).isEqualTo(result3);
+        assertThat(result4).isEqualTo(result5);
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isAfter(lastFreshTime);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        int result6 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result6).isNotEqualTo(result5);
+    }
+
+    @Test
+    public void testMaterializedViewCreatedFromTableFunctionAndTable()
+    {
+        String sourceTableName = "source_table_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + sourceTableName + " (VALUE INTEGER)");
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES 2", 1);
+        String viewName = "materialized_view_for_ptf_adn_table_" + randomNameSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM TABLE(mock.system.sequence_function()) CROSS JOIN " + sourceTableName);
+
+        List<MaterializedRow> materializedRows = computeActual("SELECT * FROM " + viewName).getMaterializedRows();
+        assertThat(materializedRows.size()).isEqualTo(1);
+        assertThat(materializedRows.get(0).getField(1)).isEqualTo(2);
+        int valueFromPtf1 = (int) materializedRows.get(0).getField(0);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
```

Comment:
Note to self: The only functionality which was unclear for me before reviewing the PR is why the `freshness` is `STALE` event after selecting from the MV.

I think it is expected that `freshness` is `UNKNOWN` until calling `REFRESH MATERIALIZED VIEW`
The MV is stale because it is not "materialized" until calling `REFRESH MATERIALIZED VIEW` so there's no storage table to load data from.